lib/Ethereum/RPC/Contract.pm

```diff
@@ -117,12 +121,21 @@ Return:
 =cut
 
 sub get_function_id {
-
-    my ($self, $function_string, @inputs) = @_;
+    my ($self, $function_string, $params_size) = @_;
+
+    my @inputs = @{$self->contract_decoded->{$function_string}};
+    my @selected_data = ();
+    if (scalar @inputs > 0) {
+        for my $v (@inputs) {
+            @selected_data = @{$v} if $params_size and @{$v} == $params_size;
+            @selected_data = @{$v} unless $params_size;
```

Comment:
This second assignment seems redundant?

Reply:
In the cases that I have some parameters this must to be the exactly size of args.
In the cases that I haven't parameters this must be considered.
But yes, i think that is better if I do that in just one line.